crypto/objects/obj_dat.c

```diff
@@ -358,23 +368,27 @@ int OBJ_obj2nid(const ASN1_OBJECT *a)
  * into an object: unlike OBJ_txt2nid it can be used with any objects, not
  * just registered ones.
  */
-
-ASN1_OBJECT *OBJ_txt2obj(const char *s, int no_name)
+static ASN1_OBJECT *ossl_obj_txt2obj(const char *s, int no_name, int lock)
 {
     int nid = NID_undef;
-    ASN1_OBJECT *op;
+    ASN1_OBJECT *op = NULL;
     unsigned char *buf;
     unsigned char *p;
     const unsigned char *cp;
     int i, j;
 
+    if (!ossl_obj_write_lock(lock))
```

Comment:
Why a write lock? Do we really need to hold a lock for this? Why can't we just let `ossl_obj_sn2nid` et al take the read locks as they need it? It seems like we're holding the lock for rather longer than we need to.

Reply:
Removed the lock and relied on the called routines taking locks individually.